main.py

```diff
@@ -101,6 +101,23 @@ def make_readme(firmwares: Iterable[Mapping[str, Any]]) -> str:
     def sort_pak_info(pak_info: Mapping[str, Any]) -> str:
         return pak_info["info"]["build_date"]
 
+    for firmware in firmwares:
+        if "sha256_pak" not in firmware:
+            if "model_id" not in firmware:
+                firmware["model_id"] = firmware["model"]
+            if "hw_ver_id" not in firmware:
+                firmware["hw_ver_id"] = firmware["hw_ver"]
+            model_title, hw_title = get_names(devices, firmware["model_id"], firmware["hw_ver_id"])
+            if model_title is None or hw_title is None:
+                print("Cannot find a match for", firmware["model_id"], '/', firmware["hw_ver_id"])
+                continue
+            model_title = model_title.strip().removesuffix("-5MP").removesuffix(" (NVR)")
+            hw_title = clean_hw_ver(hw_title)
+            firmware["build_date"] = dtparse(firmware["updated_at"]).strftime("%y%m%d")
+            firmware["download"] = [firmware["url"]]
+            firmware["info"] = firmware;
+            pak_infos.setdefault(model_title, {}).setdefault(hw_title, []).append(firmware)
```

Comment:
I get the idea of using the data sent by the API when a firmware's info is unavailable. Most of the time this would work, though Reolink has shown that they will sometimes get something wrong (see 641356e3f8a74b02ad9aa2ff7ef328e19f12f100 for example). Taking the info straight from the file means you're pretty much guaranteed that it will be the same values shown in the device's settings. For now, a PAK's info being unavailable is so rare that I'm fine not listing a firmware at all until its info is added manually. But if it were to become more frequent I would probably go with this solution too.